Framework/PythonInterface/test/python/plugins/algorithms/EnggFocusTest.py

```diff
@@ -226,6 +226,18 @@ def test_runs_ok_bank_south(self):
         self._check_output_ok(wks=out_bank_south, ws_name=out_name, y_dim_max=1201,
                               yvalues=self._expected_yvals_bank2)
 
+    def test_no_error_when_zero_proton_charge(self):
+        """
+        Run focusing on one bank but on a ws with no proton charge
+        """
+        # remove proton charge log (getProtonCharge will then return 0)
+        ws = CloneWorkspace(self.__class__._data_ws)  # so don't interfere with other tests
+        DeleteLog(Workspace=ws, Name='gd_prtn_chrg')
+        out_bank_south = EnggFocus(InputWorkspace=ws,
+                                   VanIntegrationWorkspace=self.__class__._van_integ_tbl,
+                                   VanCurvesWorkspace=self.__class__._van_curves_ws,
+                                   Bank='South', OutputWorkspace='out_bank_south')  # test passes if no RuntimeError
```

Comment:
Could move the body of this to a new function, e.g. "run_zero_proton_charge(self)", and replace the existing function with:
```
def test_no_error_when_zero_proton_charge(self):
    assertRaisesNothing(self, run_zero_proton_charge)
```
which would require `from testhelpers import assertRaisesNothing`

Reply:
Thanks for this suggestion, as discussed offline I was looking for something like this!